### PR TITLE
Solid principles

### PR DESCRIPTION
**Pull request by Deepanshu-Singh45**

### CODE REVIEW

File: Java-Assignment7/src/Part2.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            // try calling balance() on each element
+            if (cycle instanceof Unicycle) {
+                ((Unicycle) cycle).balance(); // downcasting
+            } else if (cycle instanceof Bicycle) {
+                ((Bicycle) cycle).balance(); // downcasting
+            }
+        }
+
+    }
+}
```
Comment (ShridharDaga) [inline Java-Assignment7/src/Part2.java:58]: this loop part can be place in separate class
like CycleFactory

File: Java-Assignment8/src/UserDefineException.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
+// Define three custom exceptions
+class CustomException1 extends Exception {}
+class CustomException2 extends Exception {}
+class CustomException3 extends Exception {}
+
```
Comment (ShridharDaga) [inline Java-Assignment8/src/UserDefineException.java:5]: can give some valid code what cause to throw this exception
like classic exception example AgeVerification exception (> 18 or not)

**Review by ShridharDaga — COMMENTED** [review]
work on comments

File: Java-Assignment7/src/Part2.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+public class Part2 {
+    public static void main(String[] args) {
+
+        Cycle[] cycles = {new Unicycle(1), new Bicycle(2), new Tricycle(3)};
+
+        for(Cycle cycle : cycles) {
+            cycle.ride();
+
```
Comment (ShridharDaga) [inline Java-Assignment7/src/Part2.java:48]: this loop part can be place in separate class
like CycleFactory

Comment (Deepanshu-Singh45) [reply Java-Assignment7/src/Part2.java:48]: resolved

File: Java-Core/src/AnonymousClass.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        Computer computer1 = new Computer() {
+            @Override
+            public void shutDown() {
+                System.out.println("MacBook is shutting down.");
+            }
+        };
+
+        computer1.shutDown();
+
+        Phone phone1 = new Phone() {
```
Comment (ShridharDaga) [inline Java-Core/src/AnonymousClass.java:24]: try with arrow function. Also check if an interface have more than one method , how anonymous function will work.

**Review by ShridharDaga — APPROVED** [review]

**Pull request merged**